.github/workflows/main.yml

```diff
@@ -0,0 +1,38 @@
+# Copyright 2024 Outernet Council Foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Bazel
+
+on:
+  pull_request: {}
+  push: {}
+  release:
+    types: [published]
+
+jobs:
+  build-and-test-bazel:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cache/bazelisk
+            ~/.cache/bazel
+          key: bazel-${{ hashFiles('.bazelrc', '.bazelversion', 'MODULE.bazel') }}
+          restore-keys: bazel-
+      - run: |
+          bazelisk build //...:all
+      - run: |
+          bazelisk test //...:all
```

Comment:
```suggestion
      - run: |
          bazelisk test //...:all --test_output=errors --verbose_failures --test_summary=detailed
```

Reply:
Suggesting we consider more verbose failure output, so that the contributor doesn't necessarily have to re-run these tests locally to see the error logs.

Reply:
I'll add, in case this makes outputs too verbose / hard to parse on failure - there is another option to upload the failed test logs as an GH Artifact.

Reply:
I think let's start with the test-output and we can update to having an artifact later.